tiling.py

```diff
@@ -422,29 +711,103 @@ def fundamental_domain_transforms(self):
     def center(self):
         return self._center
 
+    @property
+    def diameter(self):
+        return self._diameter
 
-class DihedralKernel(object):
-    def __init__(self, group):
-        self._group = group
+    @property
+    def radius(self):
+        return self._radius
+
+    @property
+    def group(self):
+        return self._group
+
+
+class vLineKernel(object):
+    def __init__(self, radius, center, vgroup, r2_base_kernel, flag_ceiling, flag_floor):
+        if vgroup.n not in [1]:
+            raise ValueError("Cannot construct a line kernel from planar group with dihedral "
+                             "subgroup of order not 1")
+        self._radius = radius
+        self._diameter = 2 * radius + 1
+        self._center = np.array(center)
+        self._vgroup = vgroup
+        self._r2_base_kernel = r2_base_kernel
+        self._flag_ceiling = flag_ceiling
+        self._flag_floor = flag_floor
 
     def __str__(self):
-        return "Dihedral Kernel of order %d" % self._group.n
+        return "vLine Kernel: %d by %d centered at %s" % (self._diameter, self._diameter, str(self._center))
 
     def adjacent_kernels(self, overlap):
-        return []
+
+        neighbor_directions = [(1, 0), (-1, 0)]
+        if self._flag_ceiling:
+            neighbor_directions.append((0, 1))
+
+        if self._flag_floor:
+            neighbor_directions.append((0, -1))
+
+        for neighbor in neighbor_directions:
+
+            direction_r2 = neighbor[0]
+            direction = neighbor[1]
+
+            if np.size(self._center) is not 2:
+                raise ValueError("The R3 tiling is not correct")
+            if np.size(self._r2_base_kernel.center) is not 1:
+                raise ValueError("The R2 tiling is not correct")
+
+            xCor = self._center[0] + direction_r2 * (self._r2_base_kernel.diameter - overlap)
+            yCor = self._center[1] + direction * (self._diameter - overlap)
+            new_ctr = np.array((xCor, yCor))
+            yield vLineKernel(self._radius, new_ctr, self._vgroup, self._r2_base_kernel, self._flag_ceiling, self._flag_floor)
 
     @property
     def fundamental_domains(self):
-        for k in range(len(self._group.dihedral_subgroup)):
-            transform = self._group.dihedral_subgroup[k]
-            prism = shapes.Prism(self._group.height, *self._group.fd_vertices)
-            prism.transform(transform)
-            yield (0, k), transform, prism
+        for pos, translate, _ in self.translational_fundamental_domains:
+            for k in range(len(self._vgroup.dihedral_subgroup)):
+
+                for k_r2 in range(len(self._r2_base_kernel.group.dihedral_subgroup)):
```

Comment:
What's going on here? You're trying to handle the floor and ceiling cases right? This seems way overcomplicated. Here's some quick pseudocode of how I would do this. 

```
yield (0, k, 0), transform, prism
if ceiling:
    yield (0, k, 1), transform*ceiling_reflection, transform(prism, ceiling_reflection)
elif floor:
    yield (0, k, 1), transform*floor_reflection, transform(prism, floor_reflection)
```